torch/csrc/Generator.cpp

```diff
@@ -24,22 +24,18 @@ using namespace torch;
 
 PyObject *THPGeneratorClass = nullptr;
 
-PyObject * THPGenerator_initDefaultGenerator(at::Generator* cdata)
+PyObject * THPGenerator_initDefaultGenerator(at::Generator cdata)
 {
   auto type = (PyTypeObject*)THPGeneratorClass;
   auto self = THPObjectPtr{type->tp_alloc(type, 0)};
   if (!self) throw python_error();
   auto self_ = reinterpret_cast<THPGenerator*>(self.get());
   self_->cdata = cdata;
-  self_->owner = false;
   return self.release();
 }
 
 static void THPGenerator_dealloc(THPGenerator* self)
 {
-  if (self->owner) {
-    delete self->cdata;
-  }
```

Comment:
Whoops! You're leaking the generators here. Because the PyObject now directly contains a C++ type, you must placement delete it on deallocation.

Reply:
addressed

Reply:
First, sorry, I misspoke in my comment; it's not placement delete you need to do (placement delete is not a thing), you need to just explicitly call the destructor. Assignment with the default constructor isn't sufficient, as a default constructed Generator may still have a nontrivial destructor.

```
self->cdata->~Generator();
```